ush/rocoto/workflow_utils.py

```diff
@@ -199,7 +202,7 @@ def create_wf_task(task, cdump='gdas', cycledef=None, envar=None, dependency=Non
                  'final': final}
 
     # Add PARTITION_BATCH to all non-service jobs on Orion (SLURM)
-    if get_scheduler(detectMachine()) in ['slurm'] and detectMachine() in ['ORION']:
+    if get_scheduler(detectMachine()) in ['slurm']:
```

Comment:
This line should be:
```
if get_scheduler(detectMachine()) in ['slurm'] and detectMachine() in ['ORION','JET']:
```
...otherwise the needed partition entity isn't added to each task in the xml and the jobs end up on random partitions (setting partition_batch in config.base gets ignored without this).

Reply:
In my test, the xml file contains required PARTITION_BATCH. Please see /lfs4/HFIP/hfv3gfs/Lin.Gan/expdir/test3_c192_pr/test3_c192_pr.xml

@KateFriedman-NOAA 
Please include error message or location of the file you are reporting within the message going forward.

Reply:
So yes PARTITION_BATCH is added to the top of the generated xml but the needed <partition> setting in each task section is missing without the change I requested. See this diff between two xml's I tested on Jet:

```
-bash-4.2$ pwd
/lfs4/HFIP/hfv3gfs/Kate.Friedman/expdir/pr394ffb
-bash-4.2$ diff ../pr394ff/pr394ff.xml pr394ffb.xml
...
257a258
>       <partition>&PARTITION_BATCH;</partition>
294a296
>       <partition>&PARTITION_BATCH;</partition>
333a336
>               <partition>&PARTITION_BATCH;</partition>
366a370
>       <partition>&PARTITION_BATCH;</partition>
398a403
>               <partition>&PARTITION_BATCH;</partition>
```

The pr394ff run (without `<partition>&PARTITION_BATCH;</partition>` in each batch/non-service task) submitted jobs to whichever partition was the quietist (not what we want). My pr394ffb run (with the `<partition>` settings) correctly submitted all batch jobs to PARTITION_BATCH set at the top of the xml (xjet in this case). Without the `<partition>` setting in each task in the xml the user requested partition is completely ignored.

Also, by removing the Orion part of that line you are breaking Orion support. Please make the change I requested, thanks!

Reply:
@lgannoaa 
@KateFriedman-NOAA is testing this branch on all other platforms to ensure the changes in this PR are not breaking them.
Would you like to test your PR on other platforms to make sure they don't break there?

Reply:
The test from pr394ffb and mine (/lfs4/HFIP/hfv3gfs/Lin.Gan/expdir/test2_c192_pr/test2_c192_pr.xml) are the same on job partition syntax.
The difference is pr394ffb run unnecessary steps and is using xjet and test2_c192_pr.xml is kjet. 
I recall the action of remove "and detectMachine() in ['ORION','JET']:" was requested by @aerorahul originally.

Reply:
@lgannoaa Please make the following two changes and we can be done with this discussion about partitions:

1) set partition default to 'xjet' in setup_expt_fcstonly.py (`partition_batch = 'xjet'`), users can change to kjet if they wish at setup time
2) change line in ush/rocoto/workflow_utils.py:
`if get_scheduler(detectMachine()) in ['slurm'] and detectMachine() in ['ORION','JET']:`

@aerorahul has given a thumbs up to both of these, see thumbs up emoji reactions to my requests above

Reply:
Wouldn't Lin's change do the opposite and add it to every machine that uses slurm, so it (might) break Hera rather than Orion?

That said, Hera does have a partition setting as well.